src/models/quote.js

```diff
@@ -1,38 +1,29 @@
 'use strict'
 
-const request = require('co-request')
 const UnacceptableExpiryError = require('../errors/unacceptable-expiry-error')
 const UnacceptableAmountError = require('../errors/unacceptable-amount-error')
 const AssetsNotTradedError = require('../errors/assets-not-traded-error')
-const ExternalError = require('../errors/external-error')
 const DEFAULT_DESTINATION_EXPIRY = 5 // seconds
 
 function * makeQuoteQuery (params) {
-  const sourceLedger = params.source_ledger ||
-    (yield getAccountLedger(params.source_account))
-  const destinationLedger = params.destination_ledger ||
-    (yield getAccountLedger(params.destination_account))
+  const sourceLedger = addressToLedger(params.source_address)
+  const destinationLedger = addressToLedger(params.destination_address)
   const explain = params.explain === 'true'
   return {
     sourceLedger,
     destinationLedger,
     sourceAmount: params.source_amount,
     destinationAmount: params.destination_amount,
+    destinationPrecisionAndScale: params.destination_precision && {
+      precision: params.destination_precision,
+      scale: params.destination_scale
+    },
     explain
   }
 }
 
-function * getAccountLedger (account) {
-  const res = yield request({
-    method: 'get',
-    uri: account,
-    json: true
-  })
-  const ledger = res.body && res.body.ledger
-  if (res.statusCode !== 200 || !ledger) {
-    throw new ExternalError('Unable to identify ledger from account: ' + account)
-  }
-  return ledger
+function addressToLedger (address) {
+  return address.split('.').slice(0, -1).join('.')
```

Comment:
> Ok, so if I'm understanding correctly, the actual money stops flowing once it hits the `us.fed.wf.stefan` account

From a clearing perspective, there is no such thing as "actual money". My personal ledger is a subledger of my bank's ledger which is a subledger of the Fed's ledger. If I'm transferring within my own ledger, it doesn't affect higher order ledgers. If I'm transferring within the bank's ledger, it doesn't affect higher order ledgers. If there was a subledger of my subledger and a transfer happened within that, e.g. `us.fed.wf.stefan.phone.foo` to `us.fed.wf.stefan.phone.bar`, it wouldn't appear on any higher order ledgers.

From a regulatory perspective, there *is* such a distinction ("real money" vs "pure accounting") and since regulatory issues are handled on the application layer, you may find functionality to differentiate that there.

Perhaps it's useful to show the difference between the clearing/interledger layer and application layer by example.

ILP addresses are on the Interledger layer and are used similarly to IP addresses:

ILP: `us.fed.wf.stefan`
IP: `172.217.21.174`

Note that these are low-level addresses intended primarily for use by the routing protocol. The routing protocol doesn't care about the difference between real money and fake money and so you don't see that distinction here.

On the application layer, you have human-facing identifiers:

SPSP: `alice@red.ilpdemo.org`
SMTP: `stefan@ripple.com`

Here, it's a totally different story. These are user-facing identifiers. We may need to know identities of real people for compliance purposes. I actually talked to @vhpoet and we have a tentative plan to add something exactly like you suggested `alice+phone@red.ilpdemo.org`

I think the thing that trips everybody up is that our low-level identifiers (ILP addresses) look like they're supposed to be human-readable: `us.fed.wf.bob`. We could also use some more IP address looking thing: `0123ff:213db1:123eba`. But while that would reduce some misunderstandings short-term I think in the long run it's going to make engineers' lives easier to be able to look at an ILP address and get some information from that without having to whois it.

(In my example `172.x.x.x` belongs to ARIN, i.e. it's something in North America. `172.217.x.x` belongs to Google. So it *is* possible to decode this hierarchical identifier. But because it's numbers it's a bit harder to read than, say, `na.google.web.frontend` would be.)